### PR TITLE
Deprecate register-retry-count flag from controller-manager

### PR DESCRIPTION
**Pull request by gmarek**
I don't know if we have any established policy for flag removal.

cc @bgrant0607 @thockin

### CODE REVIEW

**Comment by alex-mohr** [comment]
@davidopp for Controllers?

**Comment by bgrant0607** [comment]
We don't have a policy, but we need one.

Just ignore it and document it as deprecated. Otherwise, we break all turnup automation that used it.

**Comment by bgrant0607** [comment]
Also docs:

https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/admin/kube-controller-manager.md

I think we deleted the man pages, but there are copies in the wild:

http://rpm.pbone.net/index.php3/stat/45/idpl/29679766/numer/1/nazwa/kube-controller-manager

**Comment by gmarek** [comment]
Changed to the deprecation notice. This flag will not show in --help anymore and will print out a warning if set. I already removed it from our .md file in a previous PR.

File: cmd/kube-controller-manager/app/controllermanager.go [hunk 1]
```diff
@@ -141,6 +141,8 @@ func (s *CMServer) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&s.Master, "master", s.Master, "The address of the Kubernetes API server (overrides any value in kubeconfig)")
 	fs.StringVar(&s.Kubeconfig, "kubeconfig", s.Kubeconfig, "Path to kubeconfig file with authorization and master location information.")
 	fs.StringVar(&s.RootCAFile, "root-ca-file", s.RootCAFile, "If set, this root certificate authority will be included in service account's token secret. This must be a valid PEM-encoded CA bundle.")
+
+	fs.MarkDeprecated("register-retry-count", "This flag is currenty no-op and will be deleted.")
```
Comment (bgrant0607) [inline cmd/kube-controller-manager/app/controllermanager.go]: Could you please put this next to the flag definition?

**Comment by davidopp** [comment]
Rassigned to @bgrant0607

**Comment by bgrant0607** [comment]
LGTM

**Pull request merged**